crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1151,6 +1151,29 @@ impl<'w> EntityWorldMut<'w> {
         self
     }
 
+    /// Removes a dynamic [`Component`] from the entity if it exists.
+    ///
+    /// You should prefer to use the typed API [`EntityWorldMut::remove`] where possible.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the provided [`ComponentId`] does not exist in the [`World`].
+    pub fn remove_by_id(&mut self, component_id: ComponentId) -> &mut Self {
+        let components = &mut self.world.components;
+
+        let bundle_id = self
+            .world
+            .bundles
+            .init_component_info(components, component_id);
+
+        // SAFETY: the `BundleInfo` for this `component_id` is initialized above
+        unsafe {
+            self.remove_bundle(bundle_id);
+        }
```

Comment:
This should update the `self.location` field

Reply:
```suggestion
        self.location = unsafe { self.remove_bundle(bundle_id) };
```